src/Symfony/Component/VarDumper/VarDumper.php

```diff
@@ -115,4 +116,18 @@ private static function getDefaultContextProviders(): array
             'source' => new SourceContextProvider(null, null, $fileLinkFormatter),
         ];
     }
+
+    private static function guessMostSuitableDumper(): DataDumperInterface
+    {
+        if (\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+            return new CliDumper();
+        }
+
+        $accepted = array_map('trim', explode(',', $_SERVER['HTTP_ACCEPT'] ?? ''));
+        if (array_intersect($accepted, ['text/html', '*/*'])) {
```

Comment:
this breaks as soon as spaces are put around commas, and `q` parameters are used - I'd do just str_contains text/html instead.
this should also IMHO fallback to html when HTTP_ACCEPT is not found

Reply:
More straightforward, suits me fine.